### PR TITLE
update exhaustive to latest; use version in go.mod

### PR DESCRIPTION
**Pull request by nishanths**
This updates the exhaustive linter, and updates the dependency in go.mod to use vX.Y.Z format.

```
go get github.com/nishanths/exhaustive@v0.1.0
go mod tidy
```

Mainly address [issue #3](https://github.com/nishanths/exhaustive/issues/3).

### CODE REVIEW

**Comment by nishanths** [comment]
Tests are failing. I'll update.

File: go.mod
Comment (sayboras) [inline go.mod:40]: as you are maintainer of exhaustive, can you release tag version vX.Y.Z ? so that going forward dependenbot will just update it automatically ?

Comment (nishanths) [reply go.mod]: I am. I'll do that.

**Comment by nishanths** [comment]
This should be ready to review.

**Comment by nishanths** [comment]
Once this is merged, can I request a new release of golangci-lint? A user is affected by a false positive that is fixed by this change: https://github.com/nishanths/exhaustive/issues/3

File: .golangci.example.yml
Comment (nishanths) [inline .golangci.example.yml:106]: This is a new flag in exhaustive.

File: .golangci.yml
Comment (bombsimon) [inline .golangci.yml:13]: This is the actual file used for this project so there's no need to add defaults for visibility here.

Comment (nishanths) [reply .golangci.yml]: Thanks, updated.

File: .golangci.yml
Comment (nishanths) [inline .golangci.yml:13]: Removing because this is just the default value. As @bombsimon said, no need to mention defaults in this file.

**Review by bombsimon — APPROVED** [review]
LGTM!

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Comment by sayboras** [comment]
> Once this is merged, can I request a new release of golangci-lint? A user is affected by a false positive that is fixed by this change: [nishanths/exhaustive#3](https://github.com/nishanths/exhaustive/issues/3)

In my opinion, we are doing monthly realease, so can I wait till next week ?

**Comment by nishanths** [comment]
> In my opinion, we are doing monthly realease, so can I wait till next week ?

Yes, that works very well.

**Pull request merged**
